### ISSUE TITLE
Severely degraded performance in general

### ISSUE THREAD

**Issue by blake-sc** [suttacentral/legacy-suttacentral#76]
Performance has been degrading for several days, in terms of page load times. Pings have been fine. It came to a pass with staging nearly completely ceasing to work, followed shortly by production.

**Comment by blake-sc** [comment]
Extensive diagnosis revealed that the python servers were working fine, they produced results in milliseconds when querired over the localhost server. Once however, the connection went through apache2, something went wrong and initially it would take many seconds, or even minutes, for the cherryd servers to see the request. Eventually the requests simply stopped being served. The cherryd servers still served content fine, and so did apache as long as the proxy wasn't in play (i.e. css resources were served fine).

In the end, since production was effectively down, in desperation I restarted the vps. I then researched the proxy problem on google, and help seemed to point to KeepAlive. There were many apache processes consuming a great deal of memory, and I believe that, somehow through some pathological state in conjunction with the proxying, it simply ran out of connections, thus refusing (or deferring) incoming connections.

I have turned off KeepAlive, which should cause little harm since we serve relatively few files. Hopefully the issue is fixed, but if it's not this post should serve as some record of diagnosis on the problem.

**Comment by blake-sc** [comment]
I did another search on the KeepAlive issue:

http://abdussamad.com/archives/169-Apache-optimization:-KeepAlive-On-or-Off.html
This page conclusively states that KeepAlive increases memory use, and decreases CPU use. And the advice for our situation - tight memory, ample CPU, is to turn KeepAlive off.

In particular under Traffic Pattern the exact situation I observed occurred - the server ground to a halt when attacked by a number of web crawlers and the offline build, Apache processes started proliferating and consuming A LOT of memory. I'm still certain it involves a pathological interaction with proxying (i.e. normally KeepAlive should be fine) but with KeepAlive turned off, everything looks okay now and NewRelic has ceased to report downtime.

It seems that Apache is probably a bad choice of webserver for our purposes since it is now only being used for serving static content and proxying, which lightppd and nginx do better with less resources. However for now it isn't enough of an issue to actually do anything.

But see:
http://wiki.dreamhost.com/Web_Server_Performance_Comparison
Which has a memory use graph fitting what I saw apache doing perfectly.
